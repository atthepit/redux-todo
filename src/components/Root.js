import React from 'react'
import TodosLayout from '../containers/TodosLayout'

const Root = () =>
  <div>
    <h1>{'Hello World!'}</h1>
    <TodosLayout />
  </div>

export default Root
