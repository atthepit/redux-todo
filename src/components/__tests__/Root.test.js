import React from 'react'
import { Provider } from 'react-redux'
import Root from '../Root'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

const middlewares = []
let mockStore = undefined

beforeEach(() => {
  mockStore = configureStore(middlewares)
})


describe('Test Root component', () => {
  it('renders correctly', () => {
    const store = mockStore([
      { id: 0, text: 'Hello world', complete: false },
      { id: 1, text: 'Hello!', complete: true },
    ])

    const tree = renderer.create(
      <Provider store={ store }>
        <Root />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
