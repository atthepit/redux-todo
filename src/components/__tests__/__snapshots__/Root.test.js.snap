exports[`Test Root component renders correctly 1`] = `
<div>
  <h1>
    Hello World!
  </h1>
  <ul>
    <li
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }>
      Hello world
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "line-through",
        }
      }>
      Hello!
    </li>
  </ul>
</div>
`;
